cdmtaskservice/nersc/client.py

```diff
@@ -183,12 +184,13 @@ def _get_next_index(index: int, infinite_recovery: bool) -> int | None:
     return index if infinite_recovery else None
 
 
-async def _get_expiration(user: AsyncUser, client_id: str) -> int:
+async def _get_expiration(user: AsyncUser, client_id: str) -> datetime.datetime:
     clients = await user.clients()
     for cli in clients:
         if cli.clientId == client_id:
-            return cli.expiresAt  # TODO CLIEXPIRE check format
-    return None
-    # TODO NERSCFEATURE Go back to throwing an exception when getting the client is possible
-    #       https://nersc.servicenowservices.com/sp?id=ticket&is_new_order=true&table=incident&sys_id=714d75b8971dd210b052daa00153aff0
+            dt = datetime.datetime.fromisoformat(cli.expiresAt)
+            # TODO NERSCFEATURE ensure timezone actually is UTC
+            #      https://nersc.servicenowservices.com/sp?id=ticket&is_new_order=true&table=incident&sys_id=866f532197a51650b052daa00153affe
+            return dt.replace(tzinfo=datetime.timezone.utc)
```

Comment:
* Is it unnecessary to verify that `cli.expiresAt` is in ISO 8601 format?
* If `cli.expiresAt` contains `tzinfo`, should we use `dt.astimezone` rather than simply using replace?

It seems that `replace` only assigns the timezone info without performing a conversion to UTC.
```
In [17]: dt = datetime.datetime.fromisoformat("2024-10-29T15:30:00+05:00")

In [18]: dt.replace(tzinfo=datetime.timezone.utc)
Out[18]: datetime.datetime(2024, 10, 29, 15, 30, tzinfo=datetime.timezone.utc)

In [19]: dt.astimezone(datetime.timezone.utc)
Out[19]: datetime.datetime(2024, 10, 29, 10, 30, tzinfo=datetime.timezone.utc)

```

Reply:
> Is it unnecessary to verify that cli.expiresAt is in ISO 8601 format?

My assumption is the API will return a consistent format. If not, datetime will throw an exception which will wind up as a 500 error, which is the same thing that would happen if the code checked the format and threw an error.

> timezone

The ticket in the comment is actually asking NERSC about this. It looks to me like the timezone is UTC since it was +7 to localtime when generated, hence the replacement is just adding the missing timezone info and doesn't need a conversion. Ideally they'll just add the timezone info to to the string and I can remove the replacement. Hopefully I'll get more info tomorrow.